app/controllers/api/functions.php

```diff
@@ -865,3 +1053,141 @@
 
         $response->dynamic($execution, Response::MODEL_EXECUTION);
     });
+
+App::get('/v1/builds')
+    ->groups(['api', 'functions'])
+    ->desc('Get Builds')
+    ->label('scope', 'execution.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'listBuilds')
+    ->label('sdk.description', '/docs/references/functions/list-builds.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUILD_LIST)
+    ->param('limit', 25, new Range(0, 100), 'Results limit value. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
+    ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this param to manage pagination.', true)
+    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
+    ->param('cursor', '', new UID(), 'ID of the build used as the starting point for the query, excluding the build itself. Should be used for efficient pagination when working with large sets of data.', true)
+    ->param('cursorDirection', Database::CURSOR_AFTER, new WhiteList([Database::CURSOR_AFTER, Database::CURSOR_BEFORE]), 'Direction of the cursor.', true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function ($limit, $offset, $search, $cursor, $cursorDirection, $response, $dbForProject) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForProject */
+
+        if (!empty($cursor)) {
+            $cursorExecution = $dbForProject->getDocument('builds', $cursor);
+
+            if ($cursorExecution->isEmpty()) {
+                throw new Exception("Execution '{$cursor}' for the 'cursor' value not found.", 400);
+            }
+        }
+
+        $queries = [];
+
+        if (!empty($search)) {
+            $queries[] = new Query('search', Query::TYPE_SEARCH, [$search]);
```

Comment:
There is no search field in the data structure, do we have tests for this?

Reply:
not yet but I'll write some

Reply:
not currently, but I can write some

Reply:
Tests have been written.